### ISSUE TITLE
Tiers: Comma decimal separator fails validation in currency fields.

### ISSUE THREAD

**Issue by brymut** [opencollective/opencollective#2954]
**Describe the bug**
Using a comma as a decimal separator in currency fields doesn't work properly. Depending on browser typing a comma into a currency field either removes the whole value being typed or outright doesn't print the comma into the input.

**To Reproduce**
1. Go to the Collective page, click on the gear icon and switch to the Tiers section.
2. Add a new Tier or edit existing Tier.
2. Click on any field on with a currency symbol and try inputting an amount with a decimal using a comma as a decimal separator.
4. See error behaviour like in the screenshots below.

- The value being edited completely removed from the form

![Peek 02-03-2020 16-54](https://user-images.githubusercontent.com/1556356/75693270-fe5c2680-5ca6-11ea-95fb-335791ab13ed.gif)

- Comma not being printed into the form
![comma not recognised](https://github.com/brymut/github-gifs/blob/master/opencollective:opencollective-frontend:pull:3613/comma-not-recognised.gif?raw=true)

**Expected behaviour**
Comma accepted/saved as decimal separator 

**Additional context**
In many regions, commas are used as decimal separators in money values.

**Comment by brymut** [comment]
spotted by @Betree while reviewing https://github.com/opencollective/opencollective-frontend/pull/3613

**Comment by brymut** [comment]
Initially, I had a look around and I came across this [blog post](https://www.ctrl.blog/entry/html5-input-number-localization.html#2) on how `<input type="number">` validation is implemented differently across browser vendors and also based on the locale.
For example, on my default browser (macOS, Chrome 80),  I can't seem to be able to get a comma into the field. 
![comma not recognised](https://github.com/brymut/github-gifs/blob/master/opencollective:opencollective-frontend:pull:3613/comma-not-recognised.gif?raw=true)

However, I was able to reproduce the error behaviour you pointed out in Firefox 73 and Safari 13. 

And when I added a lang attribute (Norwegian for example) to the input element in components/InputField.js `lang=nb` commas were accepted in the input in Firefox 73 as well as periods.
![comma recognised](https://github.com/brymut/github-gifs/blob/master/opencollective:opencollective-frontend:pull:3613/comma-recognised.gif?raw=true)

But the error behaviour still persisted in the other browsers I had tried previously. 

We could try adding lang attributes that would allow commas&periods to the input fields based on the currency set, however, behaviour across browsers may not be consistent.

**Comment by Betree** [comment]
@znarf also shared something on this topic the other day: https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/. 

There are two things here:

- The question about localization, using `,` vs `.`. I've made some changes in our `StyledInputAmount`, when used as a controlled component with `parseNumbers` I believe the behavior is pretty comprehensive (but I haven't tested it on a lot of browsers):

![Peek 09-03-2020 11-25](https://user-images.githubusercontent.com/1556356/76204444-b4b89200-61f8-11ea-82c0-1b35aa68c68b.gif)

It probably won't be perfect, but even just relying on the native `input` and warning if the number is not valid is, in my opinion, an acceptable solution.

- Another problem, that is to me the really problematic part here, is that all the options after the invalid one gets removed. Let's say you have `[ 10, 20, 30, 40, 50 ]` and you try to add a comma or a dot on `30` then it should **never** remove the options after (40 and 50).

So for me the two things that we need to do here are:
- Have a clear error message if a number is not formatted properly
- If a number is invalid, don't remove the entries after it

**Comment by brymut** [comment]
I'll get on this and submit a PR asap.

> * Another problem, that is to me the really problematic part here, is that all the options after the invalid one gets removed. Let's say you have `[ 10, 20, 30, 40, 50 ]` and you try to add a comma or a dot on `30` then it should **never** remove the options after (40 and 50).

However, for this, I haven't been able to recreate this behaviour unless I'm missing something/misunderstanding you. Like in the example you shared originally it only appears to remove the invalid value being edited and doesn't affect the values after it. (like in your example, out of `[ 5, 25, 66]` when you edit the `25` that makes it invalid, the `66` seems to remain intact.)
![Peek 02-03-2020 16-54](https://user-images.githubusercontent.com/1556356/75693270-fe5c2680-5ca6-11ea-95fb-335791ab13ed.gif)

**Comment by Betree** [comment]
@brymut Yes you're right, I was not remembering the behavior correctly. So to edit my second point:
- If a number is invalid, don't remove it but eventually show a warning/an error message

**Comment by RichardLitt** [comment]
This also happens when submitting an expense.

**Comment by Betree** [comment]
This is still happening with expenses. We need to have a look at this.
